File: Algorithm/Selection_Sort.js
let selectionSort = (arr) => {
  let minIdx, temp;
  for (let i = 0; i < arr.length; i++) {
    minIdx = i;
    for (let j = i + 1; j < arr.length; j++) {
      if (arr[minIdx] > arr[j]) {
        minIdx = j;
      }
    }
    // 최솟값이 변경되면 swap
    if (minIdx !== i) {
      temp = arr[minIdx];
      arr[minIdx] = arr[i];
      arr[i] = temp;
    }
  }
  return arr;
};

let result = selectionSort([1, 3, 6, 9, 10, 4, 7, 8, 2, 5]);
console.log(result);

File: Algorithm/Selection_Sort.cpp
/***
* Selection Sort
* 시간복잡도: O(n^2)
* 최솟값을 선택해서 제일 앞으로 보내는 방식
*/
#include <stdio.h>

int main() {
  int min, temp, idx;
  int arr[10] = {4, 2, 9, 10, 1, 3, 8, 7 , 6, 5};
  for (int i = 0; i < 10; i++) {
    min = 9999;
    for (int j = i; j < 10; j++) {
      // 최솟값보다 작으면 min 교체
      if (min > arr[j]) {
        min = arr[j];
        idx = j;
      }
    }
    // swap
    temp = arr[i];
    arr[i] = arr[idx];
    arr[idx] = temp;
  }
  for (int i= 0; i < 10; i++) {
    printf("%d ", arr[i]);
  }
return 0;
}